### ISSUE TITLE
Refactor HomomorphicCommitmentFactory

### ISSUE THREAD

**Issue by SWvheerden** [tari-project/tari#610]
`HomomorphicCommitmentFactory` currently has two fn used widely:
` fn commit(&self, k: &<Self::P as PublicKey>::K, v: &<Self::P as PublicKey>::K) -> HomomorphicCommitment<Self::P>;`
and
` fn commit_value(&self, k: &<Self::P as PublicKey>::K, value: u64) -> HomomorphicCommitment<Self::P>;`

They should be be wrapped and hidden from use:
In the first one its very easy to swop the key and value around as they are the same type.
In the second one, we dont use u64 as the value type, we use MicroTari(u64) 

In short this makes 1: potentially dangerous and 2: not very useful.

**Comment by CjS77** [comment]
Done
